qemu/tests/netuser_buffer_test.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import os
+import time
+
+from avocado.utils import process
+from virttest import error_context
+from virttest import data_dir
+from virttest import utils_net
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    1) Boot guest with "-net user" and virtio-net backend
+    2) Set MTU value in guest
+    3) Compile the script and execute
+    :param test:
+    :param params:
+    :param env:
+    :return:
+    """
+    vm = env.get_vm(params["main_vm"])
+    login_timeout = params.get_numeric("login_timeout", 360)
+    vm.wait_for_login(timeout=login_timeout)
+    test_cmd = os.path.join(data_dir.get_deps_dir(), "exp.c")
+    vm.copy_files_to(test_cmd, '~')
+    vm.destroy()
+
+    params["nettype"] = "user"
+    params["vhost"] = ""
+    vm = env.get_vm(params["main_vm"])
+    params = params.object_params(vm.name)
+    vm.create(params=params)
+    login_timeout = params.get_numeric("login_timeout", 360)
+    serial_session = vm.wait_for_serial_login(timeout=login_timeout)
+
+    def mtu_test():
+        logging.info("Set mtu value and verfied")
+        serial_session.cmd(params["fw_stop_cmd"], ignore_all_errors=True)
+        guest_ifname = utils_net.get_linux_ifname(serial_session,
+                                                  vm.get_mac_address(0))
+        if guest_ifname != 'eth0':
+            test.cancel("Guest device name is not expected")
+        serial_session.cmd_output_safe(params["set_mtu_cmd"] % guest_ifname)
+        output = serial_session.cmd_output(params["check_mtu_cmd"] %
+                                           guest_ifname)
+        match_string = "mtu %s" % params["mtu_value"]
+        if match_string not in output:
+            test.fail("Guest mtu is not the expected value %s" %
+                      params["mtu_value"])
+
+    def pkg_buffer_test():
+        logging.info("Compile the script and execute")
+        serial_session.cmd("cc -xc -o ~/exp ~/exp.c")
+        serial_session.sendline(". ~/exp")
+        time.sleep(60)
+        output = process.system_output("ps -aux |grep /usr/bin/gnome-calculator",
+                                       timeout=60)
+        check_string = "/usr/bin/gnome-calculator"
+        if check_string not in output:
```

Comment:
The string will **always** be contained in the output by the given command.
```
$ ps -aux | /usr/bin/grep /usr/bin/gnome-calculator
xuhan      18452  0.0  0.0 216220   708 pts/1    S+   19:44   0:00 /usr/bin/grep /usr/bin/gnome-calculator
```

Reply:
Do you mean that the output cannot be captured?

Reply:
Hi @luckyh  I use the current method to solve the above problems, please help me review again, thanks

Reply:
Please check the debug log.
```
>>> from avocado.utils import process
>>> process.system_output("ps -aux |grep /usr/bin/gnome-calculator", timeout=60)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kar/workspace/avocado/avocado/utils/process.py", line 1167, in system_output
    encoding=encoding)
  File "/home/kar/workspace/avocado/avocado/utils/process.py", line 1037, in run
    raise CmdError(cmd, sp.result)
avocado.utils.process.CmdError: Command 'ps -aux |grep /usr/bin/gnome-calculator' failed.
stdout: b''
stderr: b"error: user name does not exist\n\nUsage:\n ps [options]\n\n Try 'ps --help <simple|list|output|threads|misc|all>'\n  or 'ps --help <s|l|o|t|m|a>'\n for additional help text.\n\nFor more details see ps(1).\n"
additional_info: None
```

Reply:
Hi @luckyh, I have changed its usage in the latest code.

Reply:
1. Add `shell=True` to avoid this ERROR.
2. `system_output` returns a bytes-like object, need to decode it.
3. Append ` | grep -v grep` to filter the output.

Reply:
Hi @PaulYuuu,  Thanks for your help, I updated the code.

Reply:
@PaulYuuu Thank you for the solution, but I think step 3 is not very appropriate in the specific use process, so I changed it according to my own ideas. If my code logic is not rigorous, please let me know.

Reply:
If guest haven't GUI, the output is:
```
ps -aux |grep /usr/bin/gnome-calculator
root     2262625  0.0  0.0   7424  1536 pts/0    S+   07:00   0:00 grep --color=auto /usr/bin/gnome-calculator
```
So we need `grep -v grep` to filter the output.
And. please clear your checkpoint.